test/unit/test_visual_recognition_v3.py

```diff
@@ -1,233 +1,717 @@
+        send_request(self, body, response)
+        assert len(responses.calls) == 1
+
+    #--------------------------------------------------------
+    # Test 2: Send only required fake data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_delete_user_data_required_response(self):
+        # Check response with required params
+        body = self.construct_required_body()
+        response = fake_response__json
+        send_request(self, body, response)
+        assert len(responses.calls) == 1
+
+    #--------------------------------------------------------
+    # Test 3: Send empty data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_delete_user_data_empty(self):
+        check_empty_required_params(self, fake_response__json)
+        check_missing_required_params(self)
+        assert len(responses.calls) == 0
+
+    #-----------
+    #- Helpers -
+    #-----------
+    def make_url(self, body):
+        endpoint = '/v3/user_data'
+        url = '{0}{1}'.format(base_url, endpoint)
+        return url
+
+    def add_mock_response(self, url, response):
+        responses.add(responses.DELETE,
+                      url,
+                      body=json.dumps(response),
+                      status=202,
+                      content_type='')
+
+    def call_service(self, body):
+        service = VisualRecognitionV3(
+            authenticator=NoAuthAuthenticator(),
+            version='2018-03-19',
+        )
+        service.set_service_url(base_url)
+        output = service.delete_user_data(**body)
+        return output
+
+    def construct_full_body(self):
+        body = dict()
+        body['customer_id'] = "string1"
+        return body
+
+    def construct_required_body(self):
+        body = dict()
+        body['customer_id'] = "string1"
+        return body
+
+
+# endregion
+##############################################################################
+# End of Service: UserData
+##############################################################################
+
+
+def check_empty_required_params(obj, response):
+    """Test function to assert that the operation will throw an error when given empty required data
+
+    Args:
+        obj: The generated test function
+
+    """
+    body = obj.construct_full_body()
+    body = {k: None for k in body.keys()}
+    error = False
+    try:
+        send_request(obj, body, response)
+    except ValueError as e:
+        error = True
+    assert error
+
+
+def check_missing_required_params(obj):
+    """Test function to assert that the operation will throw an error when missing required data
+
+    Args:
+        obj: The generated test function
+
+    """
+    body = obj.construct_full_body()
+    url = obj.make_url(body)
+    error = False
+    try:
+        send_request(obj, {}, {}, url=url)
+    except TypeError as e:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [local variable 'e' is assigned to but never used](https://app.codacy.com/manual/watson-developer-cloud/python-sdk/pullRequest?prid=4835642)